src/Simplex_tree/test/simplex_tree_extended_filtration_unit_test.cpp

```diff
@@ -0,0 +1,196 @@
+  std::clog << filt << " - " << static_cast<int>(est) << std::endl;
+  GUDHI_TEST_FLOAT_EQUALITY_CHECK(filt, copy.filtration(copy.find({5})), epsilon);
+  BOOST_CHECK(est == Gudhi::Extended_simplex_type::UP);
+
+  std::tie(filt, est) = st.decode_extended_filtration(st.filtration(st.find({6})), efd);
+  std::clog << filt << " - " << static_cast<int>(est) << std::endl;
+  BOOST_CHECK(std::isnan(filt));
+  BOOST_CHECK(est == Gudhi::Extended_simplex_type::EXTRA);
+
+  std::tie(filt, est) = st.decode_extended_filtration(st.filtration(st.find({0, 6})), efd);
+  std::clog << filt << " - " << static_cast<int>(est) << std::endl;
+  GUDHI_TEST_FLOAT_EQUALITY_CHECK(filt, copy.filtration(copy.find({0})), epsilon);
+  BOOST_CHECK(est == Gudhi::Extended_simplex_type::DOWN);
+
+  std::tie(filt, est) = st.decode_extended_filtration(st.filtration(st.find({1, 6})), efd);
+  std::clog << filt << " - " << static_cast<int>(est) << std::endl;
+  GUDHI_TEST_FLOAT_EQUALITY_CHECK(filt, copy.filtration(copy.find({1})), epsilon);
+  BOOST_CHECK(est == Gudhi::Extended_simplex_type::DOWN);
+
+  std::tie(filt, est) = st.decode_extended_filtration(st.filtration(st.find({2, 6})), efd);
+  std::clog << filt << " - " << static_cast<int>(est) << std::endl;
+  GUDHI_TEST_FLOAT_EQUALITY_CHECK(filt, copy.filtration(copy.find({2})), epsilon);
+  BOOST_CHECK(est == Gudhi::Extended_simplex_type::DOWN);
+
+  std::tie(filt, est) = st.decode_extended_filtration(st.filtration(st.find({3, 6})), efd);
+  std::clog << filt << " - " << static_cast<int>(est) << std::endl;
+  GUDHI_TEST_FLOAT_EQUALITY_CHECK(filt, copy.filtration(copy.find({3})), epsilon);
+  BOOST_CHECK(est == Gudhi::Extended_simplex_type::DOWN);
+
+  std::tie(filt, est) = st.decode_extended_filtration(st.filtration(st.find({4, 6})), efd);
+  std::clog << filt << " - " << static_cast<int>(est) << std::endl;
+  GUDHI_TEST_FLOAT_EQUALITY_CHECK(filt, copy.filtration(copy.find({4})), epsilon);
+  BOOST_CHECK(est == Gudhi::Extended_simplex_type::DOWN);
+
+  std::tie(filt, est) = st.decode_extended_filtration(st.filtration(st.find({5, 6})), efd);
+  std::clog << filt << " - " << static_cast<int>(est) << std::endl;
+  GUDHI_TEST_FLOAT_EQUALITY_CHECK(filt, copy.filtration(copy.find({5})), epsilon);
+  BOOST_CHECK(est == Gudhi::Extended_simplex_type::DOWN);
+
+  // Test to the limit - Intervals are [-2., -1.] and [1., 2.]
+  std::tie(filt, est) = st.decode_extended_filtration(Filtration_value(-2.1), efd);
+  std::clog << filt << " - " << static_cast<int>(est) << std::endl;
+  BOOST_CHECK(std::isnan(filt));
+  BOOST_CHECK(est == Gudhi::Extended_simplex_type::EXTRA);
+
+  std::tie(filt, est) = st.decode_extended_filtration(Filtration_value(2.1), efd);
+  std::clog << filt << " - " << static_cast<int>(est) << std::endl;
+  BOOST_CHECK(std::isnan(filt));
+  BOOST_CHECK(est == Gudhi::Extended_simplex_type::EXTRA);
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(simplex_tree_extend_same_filtration, Stree, list_of_tested_variants) {
```

Comment:
Could we add some calls to `decode` in this test?

Reply:
I added some tests on [025f140](https://github.com/GUDHI/gudhi-devel/pull/888/commits/025f14017de37998782aa90b7d3ea9bf0015617a)